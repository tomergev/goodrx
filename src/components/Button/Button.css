.defaultButtonStyles {
	cursor: pointer;
	color: white;
	border: none;
}