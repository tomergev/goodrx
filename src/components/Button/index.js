import React from 'react'
import './Button.css'

const Button = (props) => {
	const {
		id,
		type,
		style,
		onClick,
		children,
		className,
	} = props

	return (
		<button
			id={id}
			type={type}
			style={style}
			onClick={onClick}
			className={`defaultButtonStyles ${className}`}
		>
			{children && children}
		</button>
	)
}

export default Button