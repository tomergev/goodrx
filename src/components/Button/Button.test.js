import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Button from './index'

it('Renders without crashing', () => {
	const div = document.createElement('div')
	ReactDOM.render(<Button></Button>, div)
})

test('Creates a button', () => {
	const component = renderer.create(
		<Button onClick={() => console.log('clicked')} />
	)

	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()

	tree.props.onClick()
	tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})