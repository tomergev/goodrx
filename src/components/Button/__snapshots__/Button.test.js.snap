// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a button 1`] = `
<button
  className="defaultButtonStyles undefined"
  id={undefined}
  onClick={[Function]}
  style={undefined}
  type={undefined}
/>
`;

exports[`Creates a button 2`] = `
<button
  className="defaultButtonStyles undefined"
  id={undefined}
  onClick={[Function]}
  style={undefined}
  type={undefined}
/>
`;
